java/java.project.ui/src/org/netbeans/spi/java/project/support/ui/CreateJavaClassFileFromClipboard.java

```diff
@@ -0,0 +1,258 @@
+        StandardJavaFileManager fileManager = compiler.getStandardFileManager(null, null, null);
+
+        String publicFirstClassName = null;
+        String nonPublicFirstClassName = null;
+        String packageName = null;
+        int counter = 0;
+        JavaCompiler.CompilationTask task = compiler.getTask(null, fileManager, null, null, null, Arrays.asList(new MyFileObject(copiedData)));
+        parse:
+        try {
+            for (CompilationUnitTree compilationUnitTree : ((JavacTask) task).parse()) {
+                packageName = compilationUnitTree.getPackageName() != null
+                        ? compilationUnitTree.getPackageName().toString() : null;
+                for (Tree tree : compilationUnitTree.getTypeDecls()) {
+                    if (tree instanceof ClassTree) {
+                        final ClassTree classTree = (ClassTree) tree;
+                        if (classTree.toString().trim().startsWith(PUBLIC_MODIFIER)) {
+                            publicFirstClassName = classTree.getSimpleName().toString();
+                            break parse;
+                        }
+                        if (counter == 0) {
+                            nonPublicFirstClassName = classTree.getSimpleName().toString();
+                            counter++;
+                        }
+                    }
+                }
+            }
+        } catch (Exception ex) {
+            Exceptions.printStackTrace(ex);
+        } finally {
+            try {
+                fileManager.close();
+            } catch (IOException ex) {
+                Exceptions.printStackTrace(ex);
+            }
+        }
+
+        if (publicFirstClassName != null && !publicFirstClassName.equals("<error>")) { //NOI18N
+            return new ClassContent(publicFirstClassName, packageName);
+        } else if (nonPublicFirstClassName != null && !nonPublicFirstClassName.equals("<error>")) { //NOI18N
+            return new ClassContent(nonPublicFirstClassName, packageName);
+        } else {
+            return null;
+        }
+
+    }
+
+    private String removePackage(String copiedCode, String oldPacageName) {
+        String packageRegex = oldPacageName.replace(".", "\\s*.\\s*"); //NOI18N
+        packageRegex = "package\\s+" + packageRegex + "\\s*;"; //NOI18N
+        Pattern p = Pattern.compile(packageRegex);
+
+        Matcher m = p.matcher(copiedCode);
+        StringBuffer sb = new StringBuffer();
+        if (m.find()) {
+            m.appendReplacement(sb, ""); //NOI18N
+        }
+
+        m.appendTail(sb);
+        return sb.toString();
+    }
+
+    private String getPackageNameFromFile(File fileName) {
+        String packageLocation = null;
+        try {
+            FileObject data = FileUtil.createData(fileName);
+            JavaSource js = JavaSource.forFileObject(data);
+
+            final DeadlockTask bt = new DeadlockTask(JavaSource.Phase.RESOLVED);
+            js.runUserActionTask(bt, true);
```

Comment:
Running a task seems unnecessarily complex and slow. If I read this correctly, the goal is to get the source ClassPath. There are multiple ways to achieve that:
-ClassPath.getClassPath(ClassPath.SOURCE, data) //watch for null as a result, treat as ClassPath.EMPTY.
-ClasspathInfo.create(data).getClassPath(...SOURCE)
-js.getClasspathInfo()

The first is likely to be the fastest in this context.

Reply:
Comment addressed!
Removed the previously written code for getting class path and use the suggested one : ** ClassPath.getClassPath(data, ClassPath.SOURCE);**